buzzmobile/plan/controller/src/controller_node.cpp

```diff
@@ -0,0 +1,193 @@
+        }
+    }
+}
+
+void handleDrive(const sensor_msgs::Joy::ConstPtr& joy) {
+    //********************************************
+    //Motor control
+    float speed = 0.0;
+  
+    ROS_INFO("Last brake %d", lastBrake);
+    if (!lastBrake) {
+        if (joy->reverse_button) {
+            float correction = -1; // Normally ranges from 1 to -1. Correct so it ranges from 0 to -2
+            // Division by 2 brings to range -1 to 0. It can then be multiplied by max speed
+            speed = ((correction + joy->velocity_trigger) / 2.0) * maxFwdSpeed; 
+        } else {
+            float correction = 1; // Once inverted, will range from -1 to 1. Correct to: 0 to 2
+            // Division by 2 brings to range 0 to 1. It can then be multiplied by max speed
+            speed = ((correction + -1 * joy->velocity_trigger) / 2.0) * maxFwdSpeed;
+        }
+    }
+ 
+    ROS_INFO("Speed: %f", speed);
+    if (lastSpeed != speed) {
+        lastSpeed = speed;
+        sendMotionCommand();
+    }
+}
+
+void handleBrake(const sensor_msgs::Joy::ConstPtr& joy) {
+    int brake = 2; // 2 indicates lastBrake has not been changed;
+    if (joy->brake_button) {
+        // When we first press the button, toggle the flag and publish
+        if (!lastBrake) {
+            brake = 1;
+        }
+    } else {
+        // When we first release the button, toggle the flag and publish
+        if (lastBrake) {
+            brake = 0;
+        }
+    }
+    if (brake != 2) { // If brake was changed
+        lastBrake = brake;
+        sendMotionCommand();
+    }
+}
+
+void handleTurn(const sensor_msgs::Joy::ConstPtr& joy) {
+    float angle = 0;
+    // first test is to see if we're close to 0
+    float mag = sqrt(joy->axes[0] * joy->axes[0] + joy->axes[1] * joy->axes[1]);
+    //joystick is around the center...send 0 speed
+    if (mag > 1e-6) {
+        angle = -(atan2(fabs(joy->axes[1]), -joy->axes[0]) - M_PI_2);
+        angle = angle / (M_PI / 2); // Divide by 90 degrees to get value [0, 1]
+        angle = angle * MAX_TURNING_ANGLE; // Scale to be from [0, MAX_TURNING_ANGLE]
+        ROS_INFO("angle: [%f]", angle);
+    }
+    if (lastAngle != angle) {
+        lastAngle = angle;
+        sendMotionCommand();
+    }
+}
+
+void handleHorn(const sensor_msgs::Joy::ConstPtr& joy) {
+    lastHorn = joy->horn_button;
+}
+
+//TODO not sure what this is for. It isn't being used.
+//void honkHorn() {
+    //sound_play::SoundRequest msg;
+    //msg.sound = sound_play::SoundRequest::PLAY_FILE;
+    //msg.command = sound_play::SoundRequest::PLAY_ONCE;
+    //msg.arg = "/home/agency/Downloads/vehicle042.wav";
+    //horn_pub.publish(msg);
+//}
+
+//TODO not sure what this is for. It isn't being used
+//void obstacleCallback(const std_msgs::Bool::ConstPtr& flag) {
+//	if (!obstacleFlag && flag->data) {
+//		honkHorn();
+//	} //TODO: for some reason this will honk repeatedly (repeated obstacles)...may need to debug this
+//	
+//	obstacleFlag = flag->data;
```

Comment:
We might want another set of eyes in this file. @joshuamorton @dacohen Do you guys know what this commented out stuff does? They were in the controller in `buzzmobile-old`.